autospec/infile_handler.py

```diff
@@ -0,0 +1,138 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import re
+import requests
+import tempfile
+from urllib.request import urlretrieve
+
+import util
+import infile_bb_parser
+
+
+# filetypes and their parse order (lower parsed first)
+parsable_filetypes = {'.inc': 1, '.bb': 2}
+
+
+def parse_ext(path):
+    ext = os.path.splitext(path)[1]
+    if ext not in parsable_filetypes:
+        util.print_warning("Cannot parse infile \"{}\" type. From input: {}."
+                           " \nContinuing with the execution of "
+                           "autospec".format(ext, path))
+        return
+
+    return ext.lstrip('.')
+
+
+def check_url_content(url):
+    """
+    Check that the url to the infile file is raw or in plaintext.
+
+    This function checks the header content-type for a request to the infile
+    url. If it is html it converts eith a github url to point to the raw file,
+    a git url to point to the plaintext file, or returns None with a statement
+    that the file must be in plaintext or raw.
+    """
+    if "text/html" in requests.head(url).headers['content-type']:
+        if re.match(r'^(http)s?://github.com(.*)', url):
+            url = url.replace("github", "raw.githubusercontent").replace("blob/", "")
+        elif 'git' in url and "/tree/" in url:
+            url = url.replace("/tree/", "/plain/", 1)
+        else:
+            util.print_fatal("infile url has an html content-type, "
+                             "please use plaintext.")
+            return None
+
+    return url
+
+
+def sort_files(x):
+    return parsable_filetypes.get(os.path.splitext(x)[1])
+
+
+def parse_infile(bb_fp, output_dict, parse_type):
+    """
+    Depending on the type of file, scrape and parse the infile file returning
+    the dictionary of collected data.
+    """
+    if parse_type == "bb":
+        return infile_bb_parser.bb_scraper(bb_fp, output_dict)
+    elif parse_type == "inc":
+        return infile_bb_parser.bb_scraper(bb_fp, output_dict)
+
+
+def file_handler(indata, output_dict):
+    """
+    This function determines whether the input is a file or a url. If it is a
+    url it checks that it is in the correct format (plaintext), downloads the
+    url to a temporary file and passes the file handler to be scraped.
+    If the input is a file, then it opens the file and passes the handler to
+    be scraped.
+
+    The type of parsing bitbake, inc, deb, etc is based on the file extension.
+    """
+
+    parse_type = parse_ext(indata)
+
+    if output_dict.get('filename'):
+        output_dict['filename'].append(indata)
+    else:
+        output_dict['filename'] = [indata]
+
+    if not os.path.isfile(indata):
+        # check that input is plain or raw text and not html
+        indata = check_url_content(indata)
+        indata = None
```

Comment:
Wait, `indata` is always `None` now?

Reply:
ooo good catch this was a really bad debugging thing I left in.